app/src/main/kotlin/org/fossify/keyboard/services/SimpleKeyboardIME.kt

```diff
@@ -488,4 +493,21 @@ class SimpleKeyboardIME : InputMethodService(), OnKeyboardActionListener, Shared
 
         return Icon.createWithData(byteArray, 0, byteArray.size)
     }
+
+    // voice input
+
+    private fun getVoiceTypingIm(imm: InputMethodManager): AbstractMap.SimpleEntry<String, InputMethodSubtype>? {
+        val enabledKeyboards = imm.enabledInputMethodList
+        for (im in enabledKeyboards) for (imst in imm.getEnabledInputMethodSubtypeList(im, true))
+            if (imst.mode == "voice") return AbstractMap.SimpleEntry<String, InputMethodSubtype>(im.id, imst)
```

Comment:
Please don't do nested loops on one line, they are hard to read. Also, always use braces even if it's a one-liner.